dcicutils/s3_utils.py

```diff
@@ -293,6 +293,61 @@ def does_key_exist(self, key, bucket=None, print_error=True):
             return False
         return file_metadata
 
+    def get_object_tags(self, key, bucket):
+        """
+        Get all tags of an object.
+
+        Args:
+            key (string): object key
+            bucket (string): S3 bucket
+
+        Returns:
+            Returns list of object tags.
+            The format is [{'Key': 'KEY1','Value': 'VALUE1'},{...}]
+        """
+
+        try:
+            response = self.s3.get_object_tagging(
+                Bucket=bucket,
+                Key=key,
+            )
+            return response["TagSet"]
+        except Exception as e:
+            logger.warning(f'Could not get tags for object {bucket}/{key}: {str(e)}')
+            raise e
+
+    def set_object_tags(self, key, bucket, tags, replace_tags=True):
+        """
+        Adds or replaces tags of an object with the ones specified in `tags`.
+
+        Args:
+            key (string): object key
+            bucket (string): S3 bucket
+            tags (list): List of tags of the form [{'Key': 'KEY1','Value': 'VALUE1'}, {...}]
+            replace_tags (bool): If True, existing tags are replaced with the provided ones.
+                Otherwise, the provided tags are added to the existing ones
+
+        Returns:
+            Dict with the versionId of the object the tag-set was added to
+        """
+
+        try:
+            new_tags = tags
+            if not replace_tags:
+                existing_tags = self.get_object_tags(key, bucket)
+                if existing_tags:
+                    new_tags = existing_tags + new_tags
```

Comment:
If there are duplicate tags, it will throw an exception. I think that's better than merging and silently overwriting the old tag. It might not be intended. As stated above, the function that uses that should probably get all tags and then patch in an updated set, i.e. use `replace=True`